src/inmanta/agent/handler.py

```diff
@@ -1019,7 +1019,7 @@ def _call_discovered_resource_create_batch(
 
             discovered_resources_raw: abc.Mapping[ResourceIdStr, TDiscovered] = self.discover_resources(ctx, resource)
             discovered_resources: abc.Sequence[DiscoveredResource] = [
-                DiscoveredResource(discovered_resource_id=resource_id, values=values)
+                DiscoveredResource(discovered_resource_id=resource_id, values=values.model_dump())
```

Comment:
This is required to make pydantic accept the object used in the tests. However, the typing in here does not allow this. See this [thread](https://inmanta.slack.com/archives/CKRF0C8R3/p1697191108405509) for more details